sky/jobs/utils.py

```diff
@@ -384,8 +384,29 @@ def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
                 job_statuses = backend.get_job_status(handle, stream_logs=False)
                 job_status = list(job_statuses.values())[0]
                 assert job_status is not None, 'No job found.'
+                assert task_id is not None, job_id
+                if job_status == job_lib.JobStatus.FAILED:
+                    task_specs = managed_job_state.get_task_specs(
+                        job_id, task_id)
+                    if task_specs.get('max_restarts_on_errors', 0) == 0:
+                        # We don't need to wait for the managed job status
+                        # update, as the job is guaranteed to be in terminal
+                        # state afterwards.
+                        break
+                    print()
+                    status_display.update(
+                        ux_utils.spinner_message(
+                            'Waiting for next restart for the failed task'))
+                    status_display.start()
+                    while True:
+                        _, managed_job_status = (
+                            managed_job_state.get_latest_task_id_status(job_id))
+                        if (managed_job_status !=
+                                managed_job_state.ManagedJobStatus.RUNNING):
```

Comment:
```suggestion
                        if (managed_job_status ==
                                managed_job_state.ManagedJobStatus.RUNNING):
```
cc @Michaelvll - should this be == running so we could wait until the job starts?